stdlib/public/stubs/UnicodeNormalization.cpp

```diff
@@ -129,8 +129,8 @@ int32_t _swift_stdlib_unicode_compare_utf16_utf16(const uint16_t *LeftString,
                                                   const uint16_t *RightString,
                                                   int32_t RightLength) {
   return ucol_strcoll(GetRootCollator(),
-    LeftString, LeftLength,
-    RightString, RightLength);
+    (const UChar *)LeftString, LeftLength,
+    (const UChar *)RightString, RightLength);
```

Comment:
Without casting, compile error occurred.

In cygwin 64bit, I could get the definition with clang++ -E.
typedef wchar_t UChar;
and by simple code, sizeof(UChar) was 2.

I found where it defined.
------------
/usr/include/unicode/umachine.h

/**
 * \var UChar
 * Define UChar to be UCHAR_TYPE, if that is #defined (for example, to char16_t),
 * or wchar_t if that is 16 bits wide; always assumed to be unsigned.
 * If neither is available, then define UChar to be uint16_t.
 *
 * This makes the definition of UChar platform-dependent
 * but allows direct string type compatibility with platforms with
 * 16-bit wchar_t types.
 *
 * @stable ICU 4.4
 */
#if defined(UCHAR_TYPE)
    typedef UCHAR_TYPE UChar;
/* Not #elif U_HAVE_CHAR16_T -- because that is type-incompatible with pre-C++11 callers
    typedef char16_t UChar;  */
#elif U_SIZEOF_WCHAR_T==2
    typedef wchar_t UChar;
#elif defined(__CHAR16_TYPE__)
    typedef __CHAR16_TYPE__ UChar;
#else
    typedef uint16_t UChar;
#endif